### ISSUE TITLE
Create a Resource router component

### ISSUE THREAD

**Issue by bergie** [noflo/noflo-xpress#3]
One common use case for making servers is exposing RESTful APIs, and so the easier we can make this the better.

The resource router could be somewhat inspired by https://www.npmjs.com/package/resourcemap and expose outports for the different REST operations on a resource.

Something like:

```
'/user' -> PATH UserResource(xpress/Resource)
UserResource LIST -> REQ ListUsers # GET /user
UserResource SHOW -> REQ ShowUser # GET /user/:name
UserResource UPDATE -> REQ UpdateUser # PUT /user/:name
UserResource DELETE -> REQ DeleteUser # DELETE /user/:name
```

non-connected outports should result in either 404 or 405 depending on the case

**Comment by bergie** [comment]
Instead of giving `Resource` a path, a cleaner option might be to chain this from `Router`. Something like:

```
'all /user/*' PATTERN Router(xpress/Router)
Router REQ[0] -> REQ UserResource(xpress/Resource)
UserResource LIST -> REQ ListUsers # GET /user
...
```

**Comment by trustmaster** [comment]
Looking at the `resourcemap` examples, it seems to me that chaining with `Router` would complicate things more as `resourcemap` is applied on `app` rather than `req`. But I wonder how we could plug filters/middleware into `Resource`.

**Comment by bergie** [comment]
Yeah, Express stuff usually works via plugins. Chaining would be more noflo-y, though

**Comment by trustmaster** [comment]
Or, we could invent a Resourceful Router the FBP Way instead of trying to fit that one into the data flow.

**Comment by bergie** [comment]
For example in noflo-ui we use chained routers for flux-y events. All events from the UI go to "Dispatch" (a `GroupRouter`), which sends them to the appropriate "storage graph", where we use another `GroupRouter` to send it to the correct flow.

So packet with groups:

```
< project
< create
{ name: 'baz'}
> create
> project
```

Ends up going to the "Project" sub-router, where it gets routed to the flow handling project creation.

**Comment by bergie** [comment]
The Express 4.x `Router.route` method is also potentially interesting here:

http://expressjs.com/4x/api.html#router.route

**Comment by trustmaster** [comment]
Implemented in #70

**Comment by bergie** [comment]
Nice! Looks like a convenient way to make REST APIs with NoFlo:

![screenshot 2017-04-05 at 18 01 34](https://cloud.githubusercontent.com/assets/3346/24714898/1d7c10d8-1a2a-11e7-9bec-ee0ff9943a8b.png)
